src/MicrosoftTeams.ts

```diff
@@ -417,25 +442,132 @@ namespace microsoftTeams
+                    if (allowed)
+                    {
+                        // Open the window with a desired set of electron BrowserWindow features
+                        childWindow = currentWindow.open(link.href, "_blank", "width=" + width + ", height=" + height);
+                    }
+                    else
+                    {
+                        throw new Error("Authentication is only supported for URLs matching the pattern registered in the manifest.");
+                    }
+                };
+            }
+            else
+            {
+                // We are running in the browser so we need to center the new window ourselves
+                let left: number = (typeof currentWindow.screenLeft !== undefined) ? currentWindow.screenLeft : currentWindow.screenX;
+                let top: number = (typeof currentWindow.screenTop !== undefined) ? currentWindow.screenTop : currentWindow.screenY;
+                left += (currentWindow.outerWidth / 2) - (width / 2);
+                top += (currentWindow.outerHeight / 2) - (height / 2);
+
+                // Open the window with a desired set of standard browser features
+                childWindow = currentWindow.open(link.href, "_blank", "toolbar=no, location=yes, status=no, menubar=no, top=" + top + ", left=" + left + ", width=" + width + ", height=" + height);
+            }
+
+            if (!childWindow)
+            {
+                throw new Error("Failed to open authentication window.");
+            }
+
+            // Start monitoring the authentication window so that we can detect if it gets closed before the flow completes
+            startAuthenticationWindowMonitor();
+        }
+
+        function stopAuthenticationWindowMonitor(): void
+        {
+            if (authWindowMonitor)
+            {
+                clearInterval(authWindowMonitor);
+                authWindowMonitor = 0;
+            }
+
+            handlers["initialize"] = null;
+        }
+
+        function startAuthenticationWindowMonitor(): void
+        {
+            // Stop the previous window monitor if there is one running
+            stopAuthenticationWindowMonitor();
+
+            // Create an interval loop that:
+            // - Notifies the caller of failure if it detects that the authentication window was closed
+            // - Keeps pinging the authentication window until it establishes contact
+            authWindowMonitor = setInterval(() =>
+            {
+                if (!childWindow || childWindow.closed)
+                {
+                    handleFailure("CancelledByUser");
                 }
-                else
+                else if (!childOrigin)
                 {
-                    authenticateParameters.failureCallback(response);
+                    try
+                    {
+                        childOrigin = "*";
+                        sendMessageRequest(childWindow, "ping");
+                    }
```

Comment:
So I did at one point have logic here that stopped pinging once the childOrigin had been established; however, I found out that this approach doesn't work for VSTS since it initializes the SDK in several different pages that are part of the auth flow. Therefore, we need to keep pinging the popup until the auth flow completes and/or the popup closes.